### ISSUE TITLE
Migrate off of Stackable blocks

### ISSUE THREAD

**Issue by cjyabraham** [LF-Engineering/lfevents#578]
[Justification for removing Stackable is in this issue](https://github.com/LF-Engineering/lfevents/issues/576)

**Initial tasks**
- [x] Call with Megan to discuss how she uses the various Stackable blocks

**Develop alternative blocks and styles for each of the Stackable (ugb) blocks in use.  [Analysis is here](https://wp54-stkbl-lfeventsci.pantheonsite.io/kubecon-cloudnativecon-north-america/wp-5-4-block-analysis/).**
- [x] [ugb/divider](https://pr-580-lfeventsci.pantheonsite.io/kubecon-cloudnativecon-north-america/divider/)
- [x] [ugb/button](https://pr-580-lfeventsci.pantheonsite.io/kubecon-cloudnativecon-north-america/button-replacement/)
- [x] [ugb/blockquote](https://pr-580-lfeventsci.pantheonsite.io/kubecon-cloudnativecon-north-america/testimonial-blocks/)
- [x] [ugb/testimonial](https://pr-580-lfeventsci.pantheonsite.io/kubecon-cloudnativecon-north-america/testimonial-blocks/)
- [x] [ugb/container](https://pr-580-lfeventsci.pantheonsite.io/kubecon-cloudnativecon-north-america/container/) and [splash page example](https://pr-580-lfeventsci.pantheonsite.io/splash-page-demo/)
- [x] [ugb/icon-list](https://pr-580-lfeventsci.pantheonsite.io/kubecon-cloudnativecon-north-america/icon-list-replacement/)
- [x] [ugb/image-box](https://pr-580-lfeventsci.pantheonsite.io/kubecon-cloudnativecon-north-america/image-box/)
- [ ] [ugb/feature-grid](https://pr-580-lfeventsci.pantheonsite.io/kubecon-cloudnativecon-north-america/feature-grid-replacement/)

**Additional tasks**
- [x] There are some places in our code, [such as here](https://github.com/LF-Engineering/lfevents/blob/master/web/wp-content/themes/lfevents/page-templates/visa-request.php#L27), that use `ugb` (i.e. Stackable) styles.  We need to find them and change them for an alternative style since we will be removing all `ugb` style sheets.
- [x] Review with Megan which reusable blocks are needed and delete the others
- [x] Copy across responsive spacers so that we can pad things appropriately on mobile (James)
- [x] Deploy new block styles
- [x] Fix issues with new styles
- [x] Setup backup instance of the site so we have something to compare migrated pages to
- [x] Set out plan for migrating each of the ugb blocks to replacements
- [x] Swap out stackable on lfasiallc.com
- [x] Move all CSS from new blocks into theme css (James)
- [x] Have Data Person migrate all ugb blocks to appropriate substitute
- [x] Train Megan on using the substitutes to ugb blocks
- [x] Review all above replacement options and suggest refinements (Megan)
- [x] Deactivate and uninstall Stackable
- [x] Remove other mentions of `ugb` throughout the code, [such as here](https://github.com/LF-Engineering/lfevents/blob/master/web/wp-content/themes/lfevents/library/lfe-functions.php#L847)
- [x] Update Admin instructions
- [ ] Upgrade WP to v5.5
- [ ] Remove Find My Blocks plugin

**Comment by cjyabraham** [comment]
I've started [a PR here](https://github.com/LF-Engineering/lfevents/pull/580).

**Comment by thetwopct** [comment]
Good job on the block styles. Some initial feedback from me:

1) Container on Splash page has some issues with height on mobile - https://take.ms/EDAQL
- This is a great example why setting fixed height on the container block isn't suitable and we should not recommend it - instead to use Spacers (preferably with responsive style) to make sure there is height around the container content.

2) Dividers 
- Is there a requirement to left align / right align, not just center?

3) Feature Grid 
- On mobile, should we add more spacing between each item? This could be added as a block style, so it doesn't affect other columns blocks.

4) Featured Grid with Style
- Should there be a hover effect on these?
- Should the whole element be clickable (I can't remember exactly how Stackable did this)

5) Testimonial Blocks
- Not a fan of the lines, and them being so close to the text seems strange to me. Is this a style used elsewhere?
- The line colours seem to clash a little with the background colours, I wonder if an overlay/multiply effect could help here.
- I don't mind the second quote style, should it have rounded corners as this seems to be the style?
- General feedback - the columns have a noticeable 28px bottom margin here, preventing the quotes from being nicely centered. Perhaps this is also affecting the layout of other column blocks.
- General feedback - Quotes font sizes could do with tweaking a little on mobile, they seem big.

**Comment by cjyabraham** [comment]
1. Responsive spacers are now [in place](https://pr-580-lfeventsci.pantheonsite.io/splash-page-demo/)
2. Just center is needed for now
3. Let's see first if Megan is going to want to use this style often?
4. Sure.  We could experiment with some hover effects.  I don't think the whole element was clickable on [the Stackable examples](https://wp54-stkbl-lfeventsci.pantheonsite.io/kubecon-cloudnativecon-north-america/wp-5-4-block-analysis/).
5. I agree the lines aren't great.  For color, they should probably match the text color.  Agree on the rounded corners and your other points.  Let's get Megan's preferred style for quotes before developing this further.

**Comment by cjyabraham** [comment]
The find my blocks plugin seems to have [a fatal flaw](https://wordpress.org/support/topic/nested-blocks-3/).  We may need to consider [an alternative](https://gist.github.com/hughc/f69e46c3b624d9bb180595f5567fc654).

**Comment by cjyabraham** [comment]
Notes from @meganelizabethphillips :
- [ ] Feature grid: need to accommodate [this use-case](https://events.linuxfoundation.org/?page_id=25116) but don't have the whole box be a link; try some hover animations; could eventually turn these into block patterns
- [ ] Make button-with-expiry be able to align with the left margin of the page; currently it has some margin on its left
- [ ] Quote block: she likes the left-hand line; need to balance font size of quote with person saying it; maybe make it configurable; sometimes want to include title and company of the person quoted; needs to look good with centered text above

**Comment by cjyabraham** [comment]
I [fixed the find-my-blocks plugin](https://github.com/LF-Engineering/lfevents/commit/60ba5c4cee91862580a4338726557a62a74e156d#diff-51a1d311883aa4d9f82ec16af5aa558c) by making the block search recursive to find all nested blocks.

**Comment by cjyabraham** [comment]
Feedback from @meganelizabethphillips :
- [ ] quote and pullquote - remove dash before citation; pullquote default style keep text margins narrow in full-width style
- [ ] group - containers are used to stop speakers blocks from going full-width but instead we should have a way to set this for a speakers block so there is an option and a container isn’t needed
- [ ] feature grid - this will need to be a custom block and I've split this out into [a separate ticket](https://github.com/LF-Engineering/lfevents/issues/590)
- [ ] icon list - this will need to be a custom block and I've split this out into [a separate ticket](https://github.com/LF-Engineering/lfevents/issues/591)
- [ ] find-my-blocks doesn't include drafts and it needs to because we keep a lot of drafts to be used later

**Comment by cjyabraham** [comment]
Icon list comments:
- is there any point to the alignment selector on the block?  It's currently producing some surprising and/or unwanted effects.  Maybe just deactivate it?

**Comment by cjyabraham** [comment]
Feedback from @meganelizabethphillips :
- [ ] after deploy, turn off stackable blocks so they don't show in the chooser
- [ ] turn off regular button so people have to use button with expiry
- [ ] feature grid: try and replicate [this](https://events.linuxfoundation.org/open-source-strategy-forum/features/experiences/) and [this](https://events.linuxfoundation.org/embedded-linux-conference-europe/features/experiences/); don't have the blocks bounce
- [ ] new block: figure out solution for [this](https://events.linuxfoundation.org/embedded-linux-conference-europe/features/experiences/) and [this](https://events.linuxfoundation.org/open-source-strategy-forum/program/tracks/)
- [ ] we should improve [this table](https://events.linuxfoundation.org/kubecon-cloudnativecon-north-america/program/keynote-speaker-guide/#recording-your-keynote) when we deploy
- [ ] Make it clear in the editor when a Group block is used

**Comment by thetwopct** [comment]
Track Grid block is now uploaded. Can be created and filled in manually, but as I knew some information about the source, I spent some time making a transform which should speed up the editing process for editors. Quick video here https://cln.sh/2o4HtQ 

[Existing](https://events.linuxfoundation.org/open-networking-edge-summit-north-america/program/tracks/) - using images
[New](https://dev-lfeventsci.pantheonsite.io/open-networking-edge-summit-north-america/program/tracks/) - uses text/CSS. I'm aware that the existing design centres the last boxes if there is an uneven amount. Using CSS Grid this is not possible, but if it is a requirement I can refactor to use flex box instead.

**Comment by thetwopct** [comment]
I've spent a bit of time on the Button with Expiry block. I've made a few small changes, and added some transforms from UGB Buttons and Core Button that should make changing them over easier.

After the updates, a few blocks on my local instance needed an update - it's super simple and doesn't affect front end display. Just need to Attempt Block Recovery, like so:

![2020-11-07-230906](https://user-images.githubusercontent.com/10615884/98452312-6aecc680-214e-11eb-8500-5a8d6e4e1825.gif)
